File: 1083.java
// 1083.java

import java.util.*;
import java.io.*;

class Main {
	
	static int N, S;
	static ArrayList<Integer> lst;

	public static void main(String[] args) throws IOException {

		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer st = new StringTokenizer(br.readLine());

		N = Integer.parseInt(st.nextToken());
		st = new StringTokenizer(br.readLine());
		lst = new ArrayList<>();

		for (int i = 0; i < N; i++) {
			lst.add(Integer.parseInt(st.nextToken()));
		}
		st = new StringTokenizer(br.readLine());
		S = Integer.parseInt(st.nextToken());

		for (int i = 0; i < N - 1; i++) {
			int target = lst[i];
			int targetIdx = i;
			for (int j = i + 1; j < N; j++) {
				if (j - i > S){
					break;
				}
				if (lst[i] < lst[j]) {
					target = lst[j];
					targetIdx = j;
				}
			}
			S -= (targetIdx - i);
			lst.remove(target);
			lst.add(i, target);
		}
	}
}

File: 1083.py
# 1083.py
# 23.06.29. 10:55 ~ 11:55 ㄷㅏㅂ지

import sys
input = sys.stdin.readline

N = int(input())
lst = list(map(int, input().split()))
S = int(input())

i = 0
while i != len(lst):
	if S <= 0:
		break
	target = i
	for j in range(i + 1, i + 1 + S):
		if j == len(lst):
			break
		if lst[target] < lst[j]:
			target = j
	S -= (target-i)
	x = lst[target]
	lst.remove(x)
	lst.insert(i, x)
	i += 1

print(*lst)
